syntax = "proto3";

package auth;

option go_package = "kazali.auth.v1.;authv1";

service Auth{
  rpc Register (RegisterRequest) returns (RegisterResponse);
  rpc Login (LoginRequest) returns (LoginResponse);
  rpc IsTokenValid (IsTokenValidRequest) returns (IsTokenValidResponse);
 }

 message RegisterRequest{
  string email = 1;
  string password = 2;
  string name = 3;
 }

 message RegisterResponse{
  string user_id = 1;
 }

 message LoginRequest{
  string email = 1;
  string password = 2;
 }

 message LoginResponse{
  string token = 1;
 }

 message IsTokenValidRequest{
 }

 message IsTokenValidResponse{
  bool token_valid = 1;
 }

